txcache/selectionTracker.go

```diff
@@ -303,6 +327,7 @@ func (st *selectionTracker) getChainOfTrackedBlocks(
 		}
 
 		// extra check for a nonce gap
+		// TODO maybe add an extra check for nonce = 0
```

Comment:
Yes :+1: (left for future PR). Even though we won't be dealing genesis blocks here :see_no_evil: